src/spi/dma.rs

```diff
@@ -0,0 +1,190 @@
+use core::marker::PhantomData;
+
+use void::Void;
+
+use crate::{
+    dma::{
+        self,
+        transfer::state::{Ready, Started},
+    },
+    init_state::Enabled,
+    pac::dma0::channel::xfercfg::{DSTINC_A, SRCINC_A},
+};
+
+use super::{Instance, Master, SPI};
+
+/// An SPI/DMA transfer
```

Comment:
Maybe using `transaction` for this would be clearer, since we're not only transferring data in one direction, we're exchanging it in both

Reply:
I've been following the naming convention from `embedded-hal`: https://docs.rs/embedded-hal/0.2.4/embedded_hal/blocking/spi/trait.Transfer.html

I'm inclined to keep this as it is, but I don't have a strong opinion. I'll change the name, if you want me to.

Reply:
That's a good point, I'm inclined to leave it since the precedent is strong

Reply:
For what it's worth, I think it would be better, if `embedded-hal` called it "transaction" (and we would follow that). But that's not a battle I'm inclined to fight :-)